### PR TITLE
feat: permission class for the access of assessment

### PR DESCRIPTION
**Pull request by Kavindu-248**

### CODE REVIEW

File: apps/users/permissions.py [hunk 1]
```diff
@@ -101,3 +101,13 @@ class IsPharamcyUserOrDoctor(BasePermission):
     def has_permission(self, request, view):
         if request.method in SAFE_METHODS and request.user.role != Roles.PATIENT:
             return True
+
+class IsAllowedToAccessFormAssesment(BasePermission):
```
Comment (Raam124) [inline apps/users/permissions.py:105]: permission must be named based on user permission levels not for a specific resource

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/users/permissions.py (hunk 1 above)
Comment (Kavindu-248) [reply apps/users/permissions.py]: Name changed

File: apps/users/permissions.py [hunk 2]
```diff
@@ -101,3 +101,13 @@ class IsPharamcyUserOrDoctor(BasePermission):
     def has_permission(self, request, view):
         if request.method in SAFE_METHODS and request.user.role != Roles.PATIENT:
             return True
+
+class IsPatientOrDoctor(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.user.role == Roles.DOCTOR:
+            if obj.doctor is not None and obj.doctor != request.user:
```
Comment (Raam124) [inline apps/users/permissions.py:108]: if the doctor is none you should return those form assessments right? check this whole condition again

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/users/permissions.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/users/permissions.py]: added a condition to check when doctor is none.

File: apps/users/permissions.py [hunk 3]
```diff
@@ -101,3 +101,15 @@ class IsPharamcyUserOrDoctor(BasePermission):
     def has_permission(self, request, view):
         if request.method in SAFE_METHODS and request.user.role != Roles.PATIENT:
             return True
+
+class IsPatientOrDoctor(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.user.role == Roles.DOCTOR:
+            if obj.doctor is not None and obj.doctor != request.user.doctor:
+                return False
+            elif obj.doctor is None:
```
Comment (Raam124) [inline apps/users/permissions.py:110]: why do you need the elif block does the if block satisfy this condition as well right?

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/users/permissions.py (hunk 3 above)
Comment (Kavindu-248) [reply apps/users/permissions.py]: Noted, removed not none conditon to return. assements where doctor is null.

**Review by Raam124 — APPROVED** [review]

**Pull request merged**